com.unity.ml-agents.extensions/Documentation~/InputActuatorComponent.md

```diff
@@ -0,0 +1,42 @@
+# Integrating the Input System Package with ML-Agents
+
+## Overview
+One area we are always trying to improve is getting developers up and running with ML-Agents.  With this in mind,
+we have implemented an `InputActuatorComponent` that sets up an action space for your `Agent` based on
+an `InputActionAsset` that is referenced by the `IInputActionAssetProvider` interface, or the `PlayerInput` component
+that may be living on your player controlled `Agent`.  This means that if you have code outside of your agent that
+handles input, you will not need to implement the Heuristic function in agent as well.  The `InputActuatorComponent`
+will handle this for you.  You can now train and run inference on `Agents` with an action space defined by an `InputActionAsset`.
+
+This implementation includes:
+
+* C# `InputActuatorComponent` you can attach to your Agent.
+* An example environment where the input handling code is not in the Heuristic function of the Agent subclass.
+
+### Feedback
+We have only implemented a small subset of `InputControl` types that we thought would cover a large portion of what
+most developers would use.  Please let us know if you want more control types implemented by posting in the [ML-Agents
+forum.](https://forum.unity.com/forums/ml-agents.453/)
+
+## Getting started
+The C# code for the `InputActuatorComponent` exists inside of the extensions package (com.unity.ml-agents.extensions).  A good first step would be to familiarize with the extensions package by reading the document [here](com.unity.ml-agents.extensions.md).  The second step would be to take a look at how we have implemented the C# code in the example Input Integratin scene (located under  ML-Agents-Input-Example/Assets/ML-Agents/Examples/PushBlock/).  Once you have some familiarity, then the next step would be to add the InputActuatorComponent to your player Agent.  The example we have implemented uses C# Events to send information from the Input System.
+
+Additionally, see below for additional technical specifications on the C# code for the InputActuatorComponent.
+
+## Technical specifications for the InputActuatorComponent
+
+### `InputActuatorComponent` class
+The `InputActuatorComponent` is the bridge between ML-Agents and the Input System.. It allows ML-Agents to
+* create an `ActionSpec` for your Agent based on an `InputActionAsset`
+* send simulated input from a training process or a neural network
+* let developers keep their input handling code in one place
+
+This is accomplished by adding the `InputActuatorComponenet` to an Agent which already has the PlayerInput component attached.
```

Comment:
Is it safe so assume that this will always be on the same GameObject as the Agent, or should we allow setting it separately?

Reply:
it should always be with on the same game object as the player input component, if the agent has "use child actuators" enabled, it can be on a child game object.

Reply:
Sorry, I was being vague. The actuator definitely needs to be on the same GO as the Agent (or a child of it). But does the `PlayerInput` component? It looks like we assume that in `InputActuatorComponent.FindNeededComponents()`, but is that too strong of an assumption? For example, in a single player game, could the input component be at the root level of the scene?

Reply:
good point, it could be

Reply:
Usually, you'd want it to live on the game object you want the player to control.  This is especially true for games which have multiple players.